src/components/Oauth/OauthFlow/OauthFlow.tsx

```diff
@@ -16,10 +20,27 @@ type OauthFlowProps = {
 export function OauthFlow({
   provider, consumerRef, consumerName, groupRef, groupName,
 }: OauthFlowProps) {
-  if (GENERIC_WORKSPACE_FEATURE_FLAG && provider === PROVIDER_SALESFORCE) {
+  const apiKey = useApiKey();
+  const [providerInfo, setProviderInfo] = useState<ProviderInfo | null>(null);
+
+  useEffect(() => {
+    if (provider && api) {
+      api().providerApi.getProvider({ provider }, {
+        headers: { 'X-Api-Key': apiKey ?? '', 'Content-Type': 'application/json' },
+      }).then((_providerInfo) => {
+        setProviderInfo(_providerInfo);
+      }).catch((err) => {
+        console.error('Error loading provider info: ', err);
+      });
+    }
+  }, [apiKey, provider]);
+
+  const requiredWorkspace = providerInfo?.oauthOpts?.explicitWorkspaceRequired ?? false;
```

Comment:
either `requiresWorkspace` or `workspaceRequired` might be a better name. `requiredWorkspace` sounds odd.